target/riscv/op_helper_cheri.c

```diff
@@ -49,24 +49,35 @@
 
 enum SCRAccessMode {
     SCR_Invalid = 0,
-    ASR_Flag = 1,
-    U_Always = (PRV_U + 1) << 1,
-    U_ASR = U_Always | ASR_Flag,
-    S_Always = (PRV_S + 1) << 1,
-    S_ASR = S_Always | ASR_Flag,
-    H_Always = (PRV_H + 1) << 1,
-    H_ASR = H_Always | ASR_Flag,
-    M_Always = (PRV_M + 1) << 1,
-    M_ASR = M_Always | ASR_Flag,
+    ASR_R_Flag = 1,
+    ASR_W_Flag = 2,
+    ASR_RW_Flag = ASR_W_Flag | ASR_R_Flag,
+    U_Always = (PRV_U + 1) << 2,
+    U_ASR_W = U_Always | ASR_W_Flag,
+    U_ASR_RW = U_ASR_W | ASR_R_Flag,
+    S_Always = (PRV_S + 1) << 2,
+    S_ASR_W = S_Always | ASR_W_Flag,
+    S_ASR_RW = S_ASR_W | ASR_R_Flag,
+    H_Always = (PRV_H + 1) << 2,
+    H_ASR_W = H_Always | ASR_W_Flag,
+    H_ASR_RW = H_ASR_W | ASR_R_Flag,
+    M_Always = (PRV_M + 1) << 2,
+    M_ASR_W = M_Always | ASR_W_Flag,
+    M_ASR_RW = M_ASR_W | ASR_R_Flag,
 };
 
 static inline int scr_min_priv(enum SCRAccessMode mode)
 {
-    return ((int)mode >> 1) - 1;
+    return ((int)mode >> 2) - 1;
 }
-static inline int scr_needs_asr(enum SCRAccessMode mode)
+static inline int scr_needs_asr_r(enum SCRAccessMode mode)
```

Comment:
```suggestion

static inline int scr_needs_asr_r(enum SCRAccessMode mode)
```
to be consistent as you're touching this code